services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/extensions/dynamodb/mappingclient/functionaltests/AsyncBasicCrudTest.java

```diff
@@ -436,6 +451,17 @@ public void deleteWithConditionThatFails() {
                                                .build()).join();
     }
 
+    @Test
+    public void updateItemInterfaceToOperationSucceeds() {
+        Record record = new Record()
+            .setId("id-value")
+            .setSort("sort-value")
+            .setAttribute("one");
+
+        mappedTable.updateItem(Record.class, r -> r.item(record)).join();
+        mappedTable.updateItem(UpdateItemEnhancedRequest.builder(Record.class).item(record).build()).join();
+    }
+
```

Comment:
I think we should be able to make basic assertions here beyond 'it didn't blow up'.